docker/entrypoint.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# usage: file_env VAR [DEFAULT]
+#    ie: file_env 'XYZ_DB_PASSWORD' 'example'
+# (will allow for "$XYZ_DB_PASSWORD_FILE" to fill in the value of
+#  "$XYZ_DB_PASSWORD" from a file, especially for Docker's secrets feature)
+file_env() {
+	local var="$1"
+	local fileVar="${var}_FILE"
+	local def="${2:-}"
+	if [ "${!var:-}" ] && [ "${!fileVar:-}" ]; then
+		echo >&2 "error: both $var and $fileVar are set (but are exclusive)"
+		exit 1
+	fi
+	local val="$def"
+	if [ "${!var:-}" ]; then
+		val="${!var}"
+	elif [ "${!fileVar:-}" ]; then
+		val="$(< "${!fileVar}")"
+	fi
+	export "$var"="$val"
+	unset "$fileVar"
+}
+
+file_env 'IOTA_AUTH_USER'
+file_env 'IOTA_AUTH_PASSWORD'
+
+if [[  -z "$IOTA_AUTH_ENABLED" ]]; then
+ echo "***********************************************"
+ echo "WARNING: It is recommended to enable authentication for secure connection"
+ echo "***********************************************"
+else
+    if  [[ -z "$IOTA_AUTH_USER" ]] || [ -z "$IOTA_AUTH_PASSWORD" ]]; then
+        echo "***********************************************"
+        echo "WARNING: Default IoT Agent Auth credentials have not been overridden"
+        echo "***********************************************"
+    else
+        echo "***********************************************"
+        echo "INFO: IoT Agent Auth credentials have been overridden"
+        echo "***********************************************"
+    fi
+fi
+
+pm2 start /opt/iotaul/bin/iotagent-ul --no-daemon
```

Comment:
I hope this is correct - it seems to work but could you check?

Reply:
To run in no-daemon mode we can just run pm2-runtime. To check how to run in no-daemon mode follow https://github.com/Unitech/pm2/issues/259#issuecomment-321983223 and https://github.com/Unitech/pm2/issues/259#issuecomment-407948463

Reply:
@fgalan @Jagatjot - several options open up here we could 
- use the [PM2 Base Image](https://github.com/keymetrics/docker-pm2/blob/master/tags/8/slim/Dockerfile)
- Install as currently and run `pm2-runtime`
- Create our own `pm2.json` config for each agent
- Leave as is for now and optimise later.

I notice that we're not currently `EXPOSING` ports for `pm2` - is this deliberate? Is it something which should be added to the `Dockerfile` - or the `README` maybe?

Reply:
@dcalvoalonso was involved in the introduction of pm2 into IOTAs Dockerfiles. His opinion could be very valuable here.

Reply:
Using `pm2-runtime` seems to be the official recommendation when using Docker as it is stated in http://pm2.keymetrics.io/docs/usage/docker-pm2-nodejs/:

> Using Containers? We got your back. Start today using pm2-runtime, a perfect companion to get the most out of Node.js in a production environment.

This is also mentioned by @Jagatjot in https://github.com/telefonicaid/iotagent-ul/pull/339/files#r257553166.

Regarding

> I notice that we're not currently `EXPOSING` ports for `pm2` - is this deliberate? Is it something which should be added to the `Dockerfile` - or the `README` maybe?

Which ports are you missing @jason-fox ?

Reply:
@jason-fox For EXPOSING ports there is already an issue raised https://github.com/telefonicaid/iotagent-ul/issues/141 for which I had raised a PR https://github.com/telefonicaid/iotagent-ul/pull/342/

Reply:
@Jagatjot  - I was thinking about exposing ports needed to use Keymetrics.io - is it necessary?

https://github.com/keymetrics/docker-pm2/blob/706e19f922d75c21e60aeb817877e21497f894c1/tags/8/slim/Dockerfile#L7-L8

Reply:
The "start" part is correct? (same as https://github.com/telefonicaid/iotagent-json/pull/380#discussion_r258420973)

Reply:
Replaced `pm2 start --nodaemon` with `pm2-runtime` per recommendation.

Fixed 408ca93